### PR TITLE
Use API for "aav" command

### PR DESCRIPTION
**Pull request by XVilka**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Refactor to use API instead of `rz_core_cmd("aav")` calls where appropriate.

**Test plan**

CI is green

### CODE REVIEW

File: librz/core/canalysis.c [hunk 1]
```diff
@@ -6914,7 +6914,18 @@ RZ_IPI bool rz_core_analysis_everything(RzCore *core, bool experimental, char *d
 		rz_print_rowlog_done(core->print, oldstr);
 		rz_core_analysis_autoname_all_golang_fcns(core);
 		oldstr = rz_print_rowlog(core->print, "Analyze all flags starting with sym.go. (aF @@f:sym.go.*)");
-		rz_core_cmd0(core, "aF @@f:sym.go.*");
+		RzAnalysisFunction *fcn;
+		RzListIter *iter;
+		if (core->analysis) {
```
Comment (ret2libc) [inline librz/core/canalysis.c]: `@@f` is iteration over flags, not functions.

File: librz/core/cmd_analysis.c [hunk 2]
```diff
@@ -8203,9 +8062,10 @@ static int cmd_analysis_all(RzCore *core, const char *input) {
 	case 'd': // "aad"
 		cmd_analysis_aad(core, input);
 		break;
-	case 'v': // "aav"
-		cmd_analysis_aav(core, input);
-		break;
+	case 'v': { // "aav"
+		RzOutputMode mode = strchr(input, '*') ? RZ_OUTPUT_MODE_STANDARD : RZ_OUTPUT_MODE_RIZIN;
+		rz_core_analysis_value_pointers(core, mode);
+	} break;
```
Comment (ret2libc) [inline librz/core/cmd_analysis.c]: put the `break` inside the block, so it is properly indented at least

File: librz/core/cmd_analysis.c [hunk 3]
```diff
@@ -8203,9 +8062,10 @@ static int cmd_analysis_all(RzCore *core, const char *input) {
 	case 'd': // "aad"
 		cmd_analysis_aad(core, input);
 		break;
-	case 'v': // "aav"
-		cmd_analysis_aav(core, input);
-		break;
+	case 'v': { // "aav"
+		RzOutputMode mode = strchr(input, '*') ? RZ_OUTPUT_MODE_STANDARD : RZ_OUTPUT_MODE_RIZIN;
```
Comment (ret2libc) [inline librz/core/cmd_analysis.c]:
```suggestion
		RzOutputMode mode = strchr(input, '*') ? RZ_OUTPUT_MODE_RIZIN : RZ_OUTPUT_MODE_STANDARD;
```

**Review by ret2libc — CHANGES_REQUESTED** [review]

File: librz/core/canalysis.c (hunk 1 above)
Comment (XVilka) [reply librz/core/canalysis.c]: I checked the actual code of `@@f` foreach in `librz/core/cmd.c` and copied the logic.

Comment (ret2libc) [reply librz/core/canalysis.c]: That is the oldshell one. We merged most of `@@` and `@@@` and now there is IMO less confusion. Just do `@@?` to see what we support. In particular, you can find the `@@f:` code in `iter_flags_command`.

Comment (XVilka) [reply librz/core/canalysis.c]: Ok, removed that change from the PR, will deal with it later.

**Review by ret2libc — APPROVED** [review]

File: librz/core/canalysis.c [hunk 4]
```diff
@@ -6140,9 +6140,9 @@ static bool stringAt(RzCore *core, ut64 addr) {
 RZ_API int rz_core_search_value_in_range(RzCore *core, RzInterval search_itv, ut64 vmin,
 	ut64 vmax, int vsize, inRangeCb cb, void *cb_user) {
 	int i, align = core->search->align, hitctr = 0;
-	bool vinfun = rz_config_get_i(core->config, "analysis.vinfun");
-	bool vinfunr = rz_config_get_i(core->config, "analysis.vinfunrange");
-	bool analStrings = rz_config_get_i(core->config, "analysis.strings");
+	bool vinfun = rz_config_get_b(core->config, "analysis.vinfun");
+	bool vinfunr = rz_config_get_b(core->config, "analysis.vinfunrange");
+	bool analyzeStrings = rz_config_get_b(core->config, "analysis.strings");
```
Comment (wargio) [inline librz/core/canalysis.c]: probably this should be `analyze_strings`

Comment (XVilka) [reply librz/core/canalysis.c]: Done.

**Pull request merged**